Problem with "npm install" in `/frontend`

Seems like there's dependencies requiring attention before npm will install packages. I tried changing this package version, from an error:

        "@wagmi/core": "2.16.7",

But it seems there's more behind it. I'm looking for an example I can show some folks we're onboarding.

![Image](https://github.com/user-attachments/assets/aa2e865d-022e-4e38-869a-30d6e23b58fa)

To reproduce was `npm install` in the `/frontend` directory

---

```bash
[main]٦ npm i
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @meteorwallet/sdk@1.0.9
npm warn Found: near-api-js@5.0.0
npm warn node_modules/@near-wallet-selector/meteor-wallet/node_modules/near-api-js
npm warn   near-api-js@"5.0.0" from @near-wallet-selector/meteor-wallet@9.0.0-beta.3
npm warn   node_modules/@near-wallet-selector/meteor-wallet
npm warn     @near-wallet-selector/meteor-wallet@"9.0.0-beta.3" from the root project
npm warn
npm warn Could not resolve dependency:
npm warn peer near-api-js@"^2.0.0" from @meteorwallet/sdk@1.0.9
npm warn node_modules/@near-wallet-selector/meteor-wallet/node_modules/@meteorwallet/sdk
npm warn   @meteorwallet/sdk@"1.0.9" from @near-wallet-selector/meteor-wallet@9.0.0-beta.3
npm warn   node_modules/@near-wallet-selector/meteor-wallet
npm warn
npm warn Conflicting peer dependency: near-api-js@2.1.4
npm warn node_modules/near-api-js
npm warn   peer near-api-js@"^2.0.0" from @meteorwallet/sdk@1.0.9
npm warn   node_modules/@near-wallet-selector/meteor-wallet/node_modules/@meteorwallet/sdk
npm warn     @meteorwallet/sdk@"1.0.9" from @near-wallet-selector/meteor-wallet@9.0.0-beta.3
npm warn     node_modules/@near-wallet-selector/meteor-wallet
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: '@solana/codecs-core@2.1.0',
npm warn EBADENGINE   required: { node: '>=20.18.0' },
npm warn EBADENGINE   current: { node: 'v20.17.0', npm: '11.0.0' }
npm warn EBADENGINE }
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: '@solana/codecs-numbers@2.1.0',
npm warn EBADENGINE   required: { node: '>=20.18.0' },
npm warn EBADENGINE   current: { node: 'v20.17.0', npm: '11.0.0' }
npm warn EBADENGINE }
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: '@solana/errors@2.1.0',
npm warn EBADENGINE   required: { node: '>=20.18.0' },
npm warn EBADENGINE   current: { node: 'v20.17.0', npm: '11.0.0' }
npm warn EBADENGINE }

up to date, audited 1178 packages in 2s

271 packages are looking for funding
  run `npm fund` for details

22 vulnerabilities (2 moderate, 19 high, 1 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
```

--- a/frontend/package.json
+++ b/frontend/package.json
@@ -28,16 +28,14 @@
     "@near-wallet-selector/react-hook": "9.0.0",
     "@near-wallet-selector/sender": "9.0.0",
     "@near-wallet-selector/welldone-wallet": "9.0.0",
-    "@wagmi/connectors": "^5.7.12",
-    "@wagmi/core": "2.13.8",
-    "@web3modal/wagmi": "^5.1.11",
+    "@reown/appkit": "^1.7.3",
+    "@reown/appkit-adapter-wagmi": "^1.7.3",
     "bootstrap": "^5",
     "bootstrap-icons": "^1.11.3",
-    "near-api-js": "^5.0.1",
+    "near-api-js": "^5.1.1",
     "next": "15.2.1",
     "react": "^18",
-    "react-dom": "^18",
-    "viem": "^2.26.2"
+    "react-dom": "^18"
   },
   "devDependencies": {
     "@types/node": "^22.10.1",

--- a/frontend/src/pages/_app.js
+++ b/frontend/src/pages/_app.js
@@ -17,13 +17,13 @@ import { HelloNearContract, NetworkId } from '@/config';
 import { Navigation } from '@/components/navigation';
 
 // ethereum wallets
-import { wagmiConfig, web3Modal } from '@/wallets/web3modal';
+import { wagmiAdapter, web3Modal } from '@/wallets/web3modal';
 
 const walletSelectorConfig = {
   network: NetworkId,
   // createAccessKeyFor: HelloNearContract,
   modules: [
-    setupEthereumWallets({ wagmiConfig, web3Modal }),
+    setupEthereumWallets({ wagmiConfig: wagmiAdapter.wagmiConfig, web3Modal }),
     setupBitteWallet(),
     setupMeteorWallet(),
     setupMeteorWalletApp({contractId: HelloNearContract}),

--- a/frontend/src/wallets/web3modal.js
+++ b/frontend/src/wallets/web3modal.js
@@ -1,22 +1,46 @@
 import { injected, walletConnect } from '@wagmi/connectors';
-import { createConfig, http, reconnect } from '@wagmi/core';
-import { nearTestnet } from 'viem/chains';
-import { createWeb3Modal } from '@web3modal/wagmi';
+import { reconnect } from '@wagmi/core';
+import { createAppKit } from '@reown/appkit/react'
+import { nearTestnet } from '@reown/appkit/networks'
+import { WagmiAdapter } from '@reown/appkit-adapter-wagmi'
 
-// Get your projectId at https://cloud.reown.com
 const projectId = '5bb0fe33763b3bea40b8d69e4269b4ae';
 
-export const wagmiConfig = createConfig({
-  chains: [nearTestnet],
-  transports: { [nearTestnet.id]: http() },
-  connectors: [
-    walletConnect({ projectId, showQrModal: false }),
-    injected({ shimDisconnect: true })
-  ],
-});
+const connectors = [
+  walletConnect({
+    projectId,
+    // metadata: {
+    //   name: "NEAR Guest Book",
+    //   description: "A guest book with comments stored on the NEAR blockchain",
+    //   url: "https://near.github.io/wallet-selector",
+    //   icons: ["https://near.github.io/wallet-selector/favicon.ico"],
+    // },
+    showQrModal: false, // showQrModal must be false
+  }),
+  injected({ shimDisconnect: true }),
+];
+
+export const wagmiAdapter = new WagmiAdapter({
+  networks: [nearTestnet],
+  connectors,
+  projectId
+})
 
 // Preserve login state on page reload
-reconnect(wagmiConfig);
+reconnect(wagmiAdapter.wagmiConfig);
 
-// Modal for login
-export const web3Modal = createWeb3Modal({ wagmiConfig, projectId });
+export const web3Modal = createAppKit({
+  adapters: [wagmiAdapter],
+  projectId,
+  networks: [nearTestnet],
+  defaultNetwork: nearTestnet,
+  enableWalletConnect: true,
+  features: {
+    analytics: true,
+    swaps: false,
+    onramp: false,
+    email: false, // Smart accounts (Safe contract) not available on NEAR Protocol, only EOA.
+    socials: false, // Smart accounts (Safe contract) not available on NEAR Protocol, only EOA.
+  },
+  coinbasePreference: "eoaOnly", // Smart accounts (Safe contract) not available on NEAR Protocol, only EOA.
+});

--- a/frontend/src/wallets/web3modal.js
+++ b/frontend/src/wallets/web3modal.js
@@ -9,12 +9,6 @@ const projectId = '5bb0fe33763b3bea40b8d69e4269b4ae';
 const connectors = [
   walletConnect({
     projectId,
-    // metadata: {
-    //   name: "NEAR Guest Book",
-    //   description: "A guest book with comments stored on the NEAR blockchain",
-    //   url: "https://near.github.io/wallet-selector",
-    //   icons: ["https://near.github.io/wallet-selector/favicon.ico"],
-    // },
     showQrModal: false, // showQrModal must be false
   }),
   injected({ shimDisconnect: true }),

